### PR TITLE
feat: ensure `maxItems` with `uniqueItems` for large non-scalar arrays

### PR DESCRIPTION
**Pull request by shumkov**
BREAKING CHANGE

### CODE REVIEW

File: test/integration/dataContract/validateDataContractFactory.spec.js [hunk 1]
```diff
@@ -833,6 +833,52 @@ describe('validateDataContractFactory', () => {
         expect(error.keyword).to.equal('additionalProperties');
         expect(error.params.additionalProperty).to.equal('propertyNames');
       });
+
+      it('should have mixItems if uniqueItems is used', () => {
```
Comment (thephez) [inline test/integration/dataContract/validateDataContractFactory.spec.js]:
```suggestion
      it('should have maxItems if uniqueItems is used', () => {
```

File: test/integration/dataContract/validateDataContractFactory.spec.js [hunk 2]
```diff
@@ -833,6 +833,52 @@ describe('validateDataContractFactory', () => {
...
+        expectJsonSchemaError(result);
+
+        const [error] = result.getErrors();
+
+        expect(error.dataPath).to.equal('.documents[\'indexedDocument\'].properties[\'something\']');
+        expect(error.keyword).to.equal('required');
+        expect(error.params.missingProperty).to.equal('maxItems');
+      });
+
+      it('should have mixItems no bigger than 1000 if uniqueItems is used', () => {
```
Comment (thephez) [inline test/integration/dataContract/validateDataContractFactory.spec.js]:
```suggestion
      it('should have maxItems no bigger than 1000 if uniqueItems is used', () => {
```

File: test/integration/dataContract/validateDataContractFactory.spec.js [hunk 3]
```diff
@@ -833,6 +833,52 @@ describe('validateDataContractFactory', () => {
...
+        expectJsonSchemaError(result);
+
+        const [error] = result.getErrors();
+
+        expect(error.dataPath).to.equal('.documents[\'indexedDocument\'].properties[\'something\']');
+        expect(error.keyword).to.equal('required');
+        expect(error.params.missingProperty).to.equal('maxItems');
+      });
+
+      it('should have maxItems no bigger than 1000 if uniqueItems is used', () => {
```
Comment (thephez) [inline test/integration/dataContract/validateDataContractFactory.spec.js]: Just noticed this text is off by 2 orders of magnitude as compared with the [implementation above](https://github.com/dashevo/js-dpp/pull/108/files#diff-a4b7db40173a72aa559fe8563893e560R208)
```suggestion
      it('should have maxItems no bigger than 100000 if uniqueItems is used', () => {
```

Comment (shumkov) [reply test/integration/dataContract/validateDataContractFactory.spec.js]: Fixed it. Thank you!

**Pull request merged**